modelcontextprotocol/tools/search.py

```diff
@@ -146,18 +151,27 @@ def search_assets(
                 if isinstance(condition, dict):
```

Comment:
We've defined input condition as dict, so shouldn't it alway be a dictionary? Also we should remove other forms of condiftion and return an error in those cases